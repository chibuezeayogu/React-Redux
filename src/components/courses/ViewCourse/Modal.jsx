import React, { Component } from 'react';
import {
	func, string, array, bool, shape 
} from 'prop-types';

const Modal = ({
	closeModal, deleteCourse, modalContent, displayModal
}) => (
	<div 
		className="modal" 
		id="delete-course-modal"
		style={(displayModal) ? { display: 'block' } : { display: 'none' }}
	>
		<div className="modal-content">
			<div className="modal-header">
				<div className="header-title">Delete Course</div>
			</div>
			<h3>{'Are you sure you want to delete'}</h3>
			<span className="warning">{modalContent.title}</span>
			<div className="modal-footer">
				<div className="">
					<button 
						className="grayed upper" 
						type="button"
						onClick={closeModal}
					>
                Cancel
					</button>
					<button
						className="fill upper delete-course" 
						type="button" 
						tabIndex={0}
						onClick={() => deleteCourse(modalContent.id)}
					>
                Delete
					</button>
				</div>
			</div>
		</div>
	</div>
);

Modal.propTypes = {
	closeModal: func,
	displayModal: bool,
	deleteCourse: func,
	modalContent: shape({
		id: string,
		title: string,
	})
};

export default Modal;