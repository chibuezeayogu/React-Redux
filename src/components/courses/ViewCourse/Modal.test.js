import React from 'react';
import { shallow } from 'enzyme';

import Modal from './Modal';

describe('Modal Component', () => {
	const props = {
		modalContent: {
			id: 'react-flux-building-applications',
			title: 'Building Applications in React and Flux',
			watchHref: 'http://www.pluralsight.com/courses/react-flux-building',
			authorId: 'cory-house',
			length: '5:08',
			category: 'JavaScript'
		},
		deleteCourse: jest.fn(),
		closeModal: jest.fn(),
		displayModal: true
	};
  
	const wrapper = shallow(<Modal {...props} />);
  
	it('should render correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});
});