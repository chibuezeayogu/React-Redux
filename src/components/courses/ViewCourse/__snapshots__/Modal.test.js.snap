// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should render correctly 1`] = `
<div
  className="modal"
  id="delete-course-modal"
  style={
    Object {
      "display": "block",
    }
  }
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <div
        className="header-title"
      >
        Delete Course
      </div>
    </div>
    <h3>
      Are you sure you want to delete
    </h3>
    <span
      className="warning"
    >
      Building Applications in React and Flux
    </span>
    <div
      className="modal-footer"
    >
      <div
        className=""
      >
        <button
          className="grayed upper"
          onClick={[MockFunction]}
          type="button"
        >
          Cancel
        </button>
        <button
          className="fill upper delete-course"
          onClick={[Function]}
          tabIndex={0}
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
